src/config_trapdoor.c

```diff
@@ -1968,6 +1969,351 @@ TbBool make_available_all_traps(PlayerNumber plyr_idx)
+    switch (property)
+    {
+        case 2: // NametextId
+            doorst->name_stridx = value;
+            break;
+        case 3: // TooltipTextId
+            doorst->tooltip_stridx = value;
+            manufctr->tooltip_stridx = doorst->tooltip_stridx;
+            update_trap_tab_to_config();
+            break;
+        case 4: //SymbolSprites
+            {
+                doorst->bigsym_sprite_idx = value; // First
+                doorst->medsym_sprite_idx = value2; // Second
+                manufctr->bigsym_sprite_idx = doorst->bigsym_sprite_idx;
+                manufctr->medsym_sprite_idx = doorst->medsym_sprite_idx;
+                update_trap_tab_to_config();
+            }
+            break;
+        case 5: // PointerSprites
+            doorst->pointer_sprite_idx = value;
+            update_trap_tab_to_config();
+            break;
+        case 6: // PanelTabIndex
+            doorst->panel_tab_idx = value;
+            manufctr->panel_tab_idx = value;
+            update_trap_tab_to_config();
+            break;
+        case 7: // Crate
+            game.conf.object_conf.object_to_door_or_trap[value] = door_type;
+            game.conf.object_conf.workshop_object_class[value] = TCls_Door;
+            game.conf.trapdoor_conf.door_to_object[door_type] = value;
+            break;
+        case 8: // ManufactureLevel
+            doorst->manufct_level = value;
+            break;
+        case 9: // ManufactureRequired
+            doorst->manufct_required = value;
+            break;
+        case 10: // Health
+            if (door_type < game.conf.trapdoor_conf.door_types_count)
+            {
+                doorst->health = value;
+            }
+            update_all_door_stats();
+            break;
+        case 11: // SlabKind
+            if (door_type < game.conf.trapdoor_conf.door_types_count)
+            {
+                doorst->slbkind[0] = value2;
+                doorst->slbkind[1] = value;
+            }
+            update_all_door_stats();
+            break;
+        case 12: // OpenSpeed
+            if (door_type < game.conf.trapdoor_conf.door_types_count)
+            {
+                doorst->open_speed = value;
+            }
+            break;
+        case 13: // Properties
+            doorst->model_flags = value;
+            break;
+        case 14: //SellingValue
+            doorst->selling_value = value;
+            break;
+        case 15: // Unsellable
+            doorst->unsellable = value;
+            break;
+        case 16: // PlaceSound
+            if (door_type < game.conf.trapdoor_conf.door_types_count)
+            {
+                doorst->place_sound_idx = value;
+            }
+            break;
+        default:
+            WARNMSG("Unsupported Door configuration, variable %d.", property);
+            break;
+    }
+}
+
+void script_set_trap_configuration(ThingModel trap_type, short property, long value, long value2, long value3, long value4)
```

Comment:
Why are you moving these script specific functions out of the script related c-files? You had the script_lib for that right?

Reply:
I'd say everything related to parsing scripts, and determining what logic should be run and when as script related, assigning values to config fields is more config related logic

Reply:
script_process_new_object and script_process_new_effectgen should probably be moved as well, but the others are related to parsing the script

Reply:
hmm yeah but where is a good spot ppl would actually read it?

Reply:
that initial split was also my first PR on fx so was mostly splitting some stuff up to start and make sense of it, as files that are many thousands of lines long are very hard to understand,
script_lib mostly contains utility functions the script commands use to help them parse the scripts

parsing ScriptLine to a ScriptContext and then reparsing that ScriptContext to function parameters logic further down would take is all script related, 
script_set_trap_configuration updates configs, updating configs in itself isn't directly script related, it's just called by script related files, so why would it be in script related files

Reply:
Fine, I'll do a quick smoke test later and then merge. But I do believe script_lib was meant for that, which is why such a 'purpose' comment is quite nice to have. The script_lib file seems mostly unused now anyway.